dask/dataframe/groupby.py

```diff
@@ -1143,6 +1143,7 @@ def _aca_agg(
         token,
         func,
         aggfunc=None,
+        metafunc=None,
```

Comment:
Making a change like this is concerning.  Why was this necessary?

Reply:
If I use a function such as `M.head` for example then the grouping keys are not propagate to the `aggregate_func` (because [the pandas operation does not return the grouping keys](https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.GroupBy.head.html)). Therefore I created a special function to be applied to the chunks: `_head_chunk`. This function returns a MultiIndex, if I use this function to compute the meta info then the index dtype wont match the pandas one. That is the reason that I decided to use two separate functions for computing the meta and applying to chunks. I could create a new function and leave `_aca_agg` unmodified if that is the concern. Any other other alternatives?

Reply:
It would probably be more palatable to include an optional `meta` kwarg that accepts a precomputed `meta` rather than `metafunc`.

Reply:
@jsignell Yes, you are right.